discovery-provider/src/tasks/index.py

```diff
@@ -292,17 +292,23 @@ def fetch_ipfs_metadata(
                 if cid_type[cid] == "track"
                 else user_metadata_format
             )
-            user_replica_set = user_to_replica_set[
-                cid_to_user_id[cid]
-            ]  # user or track owner's replica set
-            future = executor.submit(
-                update_task.ipfs_client.get_metadata,
-                cid,
-                metadata_format,
-                user_replica_set,
-            )
-            futures.append(future)
-            futures_map[future] = [cid, txhash]
+            user_id = cid_to_user_id[cid]
+            if user_id in user_to_replica_set:
```

Comment:
i think we should check if cid in cid_to_user_id in the if check as well

Reply:
good catch! thanks